Feature/homework3

Ringonoringo opened the pull request.

Commit:
3дз обозначила все методы.
задание1 готово
задание 2 готово. может надо использовать массив?
задание 3 готово

Commit:
задание 1 готово

Commit:
5 заданий готовы.первый возможный вариант.

Commit:
причесывание

FlameXander commented on src/main/java/ru/nabuhiro/java/basic/homeworks/lesson7/TwoDimensionalMassive.java:
```diff
@@ -0,0 +1,84 @@
+
+    //Реализовать метод findMax(int[][] array) который должен найти и вернуть максимальный элемент массива;
+    public static int findMax(int[][] array) {
+        int maxValue = 0;
```
Если массив будет состоять полностью из отрицательных чисел, то решение будет немного некорректным, т.е. вернется 0, надо бы поправить

FlameXander commented on src/main/java/ru/nabuhiro/java/basic/homeworks/lesson7/TwoDimensionalMassive.java:
```diff
@@ -0,0 +1,84 @@
+
+    //Реализуйте метод, который считает сумму элементов второй строки двумерного массива, если второй строки не существует,
+// то в качестве результата необходимо вернуть -1
+    public static int twoStringSum(int[][] array) {
```
А проверяли работу этого метода? Пробовали передать ему массив из трех строк? Получится ведь в результате -1? Надо вначале сделать проверок что строк не меньше двух, а потом одинарным циклом можно посчитать сумму второй строки

Commit:
задание 4. присвоение вместо 0, значения первого элемента массива.
задание 5. массив двумерный но строка 1. подбор вида массива изначального и проверка. упрощение цикла.

FlameXander commented on src/main/java/ru/nabuhiro/java/basic/homeworks/lesson7/TwoDimensionalMassive.java:
```diff
@@ -69,15 +75,15 @@ public static int findMax(int[][] array) {
 // то в качестве результата необходимо вернуть -1
     public static int twoStringSum(int[][] array) {
         int sumElements = 0;
```
Я бы рекомендовал упростить код до:
```
public static int twoStringSum(int[][] array) {
    if (array.length < 2) {
        return -1;
    }        
    int sumElements = 0;
    for (int j = 0; j < array.length ; j++) {
        sumElements += array[1][j];
    }
    return sumElements;
}
```
